substrate/client/network/sync/src/engine.rs

```diff
@@ -478,6 +477,12 @@ where
 		let max_out_peers = net_config.network_config.default_peers_set.out_peers;
 		let max_in_peers = (max_full_peers - max_out_peers) as usize;
 
+		let tick_timeout = {
+			let mut interval = tokio::time::interval(TICK_TIMEOUT);
+			interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+			interval
```

Comment:
This is a tokio [`interval`](https://docs.rs/tokio/latest/tokio/time/fn.interval.html) timer, it ticks automatically every `TICK_TIMEOUT`. No need to reset it every time.

Here we set the missed tick behavior so that if the tick is missed (due to time spent processing other futures), the timer expires `TICK_TIMEOUT` seconds after [`tick()`](https://docs.rs/tokio/latest/tokio/time/struct.Interval.html#method.tick) is actually called, instead of firing in burst for every missed tick (the default).